cluster-configs.md

```diff
@@ -0,0 +1,110 @@
+# Cluster configs in the oshinko REST api
+
+## Introduction
+
+Currently the oshinko REST api allows Spark node counts
+to be set when the cluster is created, but cluster
+configurations will likely be more complex in the future.
+This specification proposes that cluster configuration
+settings be logically grouped in REST objects and that
+named configurations be added as a feature.
+
+## Problem statement
+
+The initial oshinko REST api presents a very simple view of
+cluster configuration: a user may specify the number of master
+nodes and the number of worker nodes with fields in the REST
+object (note however that at this time master count > 1 is meaningless).
+
+As oshinko matures, cluster configurations may expand
+to include things like HA master configuration, pod resource
+limits, or other settings (disclaimer, not intended as a roadmap).
+It makes sense to group cluster configuration settings logically
+in the REST objects rather than include them as unrelated separate
+fields.
+
+Additionally, the ability to reference named configurations
+as an alternative to explicit settings in a cluster object
+would be a benefit to users in general and novice users in
+particular. Given logical groupings, named configurations
+become a possibility.
+
+## Proposed solution
+
+The master and worker count fields in any current REST objects
+should be replaced with a configuration object that contains
+both fields and may in the future contain other fields. For
+example, a JSON cluster creation object would like like this:
+
+    {
+      "name": "fred",
+      "config":
+              {
+	        "masterCount": 1,
+		"workerCount": 2
+	      }
+    }
+			   
+A `namedConfig` field should be added that
+can be used to refer to a stored configuration by name.
+For example:
+
+    {
+      "name": "fred",
+      "namedConfig": "small",
```

Comment:
i was thinking something along the line of just having the name inside the config, if we don't allow updating it will unambiguous. for example:
```
{
  "name": "fred",
  "config": {
    "name": "small",
    "workerCount": 5           <--- this would be a modification on the small configuration
  }
```